src/com/tapia/foss/frames/Visualizer.java

```diff
@@ -210,8 +210,14 @@ private void jButtonRemoveActionPerformed(java.awt.event.ActionEvent evt) {//GEN
                 JOptionPane.showMessageDialog(this, "Please enter a value to be removed");
             else
             {
-                if(!structure.remove(Integer.parseInt(input)))
-                    JOptionPane.showMessageDialog(this, "Unable to find/delete "+input);
+                
+                if (this.tryParsing(input)) {
+                    if(!structure.remove(Integer.parseInt(input))) {
+                        JOptionPane.showMessageDialog(this, "Unable to find/delete "+input);
+                    }
+                } else {
+                    return;
```

Comment:
Instead of returning, how about we give a message to the user using JOptionPane giving feedback on sending in an integer only.
I dont think return is necessary here, is it?